### PR TITLE
Refactor: Isolate domain watcher

### PR DESCRIPTION
**Pull request by xpivarc**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/virt-handler/cache/domain-watcher.go [hunk 1]
```diff
@@ -0,0 +1,319 @@
+package cache
```
Comment (fossedihelm) [inline pkg/virt-handler/cache/domain-watcher.go:1]: missing header

File: pkg/virt-handler/cache/cache.go [hunk 2]
```diff
@@ -251,8 +251,8 @@ func listSockets() ([]string, error) {
 	return sockets, nil
 }
 
-func NewSharedInformer(virtShareDir string, watchdogTimeout int, recorder record.EventRecorder, vmiStore cache.Store, resyncPeriod time.Duration) (cache.SharedInformer, error) {
+func NewSharedInformer(virtShareDir string, watchdogTimeout int, recorder record.EventRecorder, vmiStore cache.Store, resyncPeriod time.Duration) cache.SharedInformer {
 	lw := newListWatchFromNotify(virtShareDir, watchdogTimeout, recorder, vmiStore, resyncPeriod)
 	informer := cache.NewSharedInformer(lw, &api.Domain{}, 0)
-	return informer, nil
+	return informer
```
Comment (fossedihelm) [inline pkg/virt-handler/cache/cache.go:257]: you can return directly:
```
return cache.NewSharedInformer(lw, &api.Domain{}, 0)
```

**Review by fossedihelm — COMMENTED** [review]
Thank you! Some comments below.
Also, since the commits are 1 month old maybe it is worth it to rebase on main

**Comment by enp0s3** [comment]
/approve
@xpivarc Thanks for the effort! once comments addressed we can get it in.

File: pkg/virt-handler/cache/cache.go (hunk 2 above)
Comment (xpivarc) [reply pkg/virt-handler/cache/cache.go]: Done

File: pkg/virt-handler/cache/domain-watcher.go (hunk 1 above)
Comment (xpivarc) [reply pkg/virt-handler/cache/domain-watcher.go:19]: done

**Review by fossedihelm — APPROVED** [review]

**Pull request merged**